addons/web/static/tests/core/owl_popover_tests.js

```diff
@@ -0,0 +1,332 @@
+                    assert.step(
+                        this.state.showPopover ? 'popover_hide' : 'popover_show'
+                    );
+                    this.state.showPopover = !this.state.showPopover;
+                }
+            }
+            Parent.components = { Popover, SubComponent };
+            Parent.env = makeTestEnvironment();
+            Parent.template = xml`
+                <div>
+                    <button
+                        id="target"
+                        t-on-click="_onTogglePopover"
+                        >
+                        Notice me, senpai
+                    </button>
+                    <br/>
+                    <span id="passiveTarget">😭</span>
+                    <Popover
+                        t-if="state.showPopover"
+                        position="state.position"
+                        title="state.title"
+                        target="state.target"
+                        t-on-close-popover="_onTogglePopover"
+                        >
+                        <SubComponent text="state.textContent"/>
+                    </Popover>
+                </div>`;
+
+            const parent = new Parent();
+            const fixture = testUtils.prepareTarget(true);
+
+            // Helper functions
+            async function changeProps(key, value) {
+                parent.state[key] = value;
+                await testUtils.nextTick();
+            }
+            function pointsTo(popover, element, position) {
+                const hasCorrectClass = popover.classList.contains(
+                    `o_popover--${position}`
+                );
+                const expectedPosition = Popover._computePositioningData(
+                    popover,
+                    element
+                )[position];
+                const correctLeft =
+                    parseFloat(popover.style.left) ===
+                    Math.round(expectedPosition.left * 100) / 100;
+                const correctTop =
+                    parseFloat(popover.style.top) ===
+                    Math.round(expectedPosition.top * 100) / 100;
+                return hasCorrectClass && correctLeft && correctTop;
+            }
+
+            // set debug because popovers adapt their position based on
+            // the viewport; if the fixture is outside the viewport
+            // half the tests will fail
+            await parent.mount(fixture);
+            /* shift the component some amount to the right to avoid
+             triggering the code that prevents having a popover outside
+             the view port; the popover should fit under but not to the
+             left */
+            parent.el.style.marginLeft = '150px';
+
+            const body = document.querySelector('body');
+            let popover, element, title;
+            // Show/hide
+            assert.containsNone(
+                body,
+                '.o_popover',
+                'There should be no popover'
```

Comment:
We don't really put error message nowadays. the assert is considered meaningful enough

Reply:
also, this fits in a single line

Reply:
that's just prettier doing it's job :)